lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -0,0 +1,176 @@
+    }
+
+    return eventSourceArn.split('/')[1];
+  }
+
+  compileMSKEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      if (functionObj.events) {
+        // It is required to add the following statement in order to be able to connect to MSK cluster
+        const ec2Statement = {
+          Effect: 'Allow',
+          Action: [
+            'ec2:CreateNetworkInterface',
+            'ec2:DescribeNetworkInterfaces',
+            'ec2:DescribeVpcs',
+            'ec2:DeleteNetworkInterface',
+            'ec2:DescribeSubnets',
+            'ec2:DescribeSecurityGroups',
+          ],
+          Resource: '*',
+        };
+        const mskStatement = {
+          Effect: 'Allow',
+          Action: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers'],
+          Resource: [],
+        };
+
+        functionObj.events.forEach(event => {
+          if (event.msk) {
+            const EventSourceArn = event.msk.arn;
+            const Topic = event.msk.topic;
+            const BatchSize = event.msk.batchSize || 100;
+            const Enabled = event.msk.enabled != null ? event.msk.enabled : true;
+            const StartingPosition = event.msk.startingPosition || 'TRIM_HORIZON';
+
+            const mskClusterName = this.getMSKClusterName(EventSourceArn);
+            const mskEventLogicalId = this.provider.naming.getMSKEventLogicalId(
+              functionName,
+              mskClusterName,
+              Topic
+            );
+
+            const funcRole = functionObj.role || this.serverless.service.provider.role;
+            const dependsOn = this.resolveDependsOn(funcRole);
+
+            const lambdaLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
+
+            const mskResource = {
+              Type: 'AWS::Lambda::EventSourceMapping',
+              DependsOn: dependsOn,
+              Properties: {
+                BatchSize,
+                EventSourceArn,
+                FunctionName: {
+                  'Fn::GetAtt': [lambdaLogicalId, 'Arn'],
+                },
+                StartingPosition,
+                Enabled,
+                Topics: [Topic],
+              },
+            };
+
+            mskStatement.Resource.push(EventSourceArn);
+
+            const newMSKObject = {
+              [mskEventLogicalId]: mskResource,
+            };
+
+            Object.assign(cfTemplate.Resources, newMSKObject);
+          }
+        });
+
+        if (cfTemplate.Resources.IamRoleLambdaExecution) {
+          const statement =
+            cfTemplate.Resources.IamRoleLambdaExecution.Properties.Policies[0].PolicyDocument
+              .Statement;
+          if (mskStatement.Resource.length) {
+            statement.push(mskStatement);
+            statement.push(ec2Statement);
+          }
+        }
```

Comment:
We should add it only if at least one `msk` event was configured. Here if I see correctly it's added unconditionally

Reply:
It should work correctly unless I'm missing something - I also had a unit test for it to ensure it's not altering the role if there are not `msk` events. It's checking for `mskStatement.Resource.length` - if it's non-zero, that means we have at least one `msk` event. I followed a similar approach as in `stream` and `sqs` here.

Reply:
Indeed, I didn't notice `if (mskStatement.Resource.length)` condition. 👍